File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.py
N = int(input())

data = list(map(int, input().split()))
dp = [0] * N

ans = 0
for i in range(1, N):
    for j in range(0, i):
        if data[i] > data[j]:
            dp[i] = max(dp[i], dp[j] + 1)
            ans = max(ans, dp[i])
print(ans + 1)

File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.cc
#include <cstdio>
#include <algorithm>
using namespace std;
#pragma warning(disable:4996)

int main() {

	int data[1001];
	int dy[1001] = { 0, };
	int n;
	int Max = 0;
	scanf("%d", &n);
	for (int i = 0; i < n; i++) scanf("%d", &data[i]);
	for (int i = 1; i < n; i++) {
		for (int j = 0; j < i; j++) {
			if (data[j] < data[i]) dy[i] = max(dy[j] + 1, dy[i]);
			Max = max(Max, dy[i]);
		}
	}
	printf("%d", Max + 1);
}